maven_iuvs/echelle.py

```diff
@@ -2070,24 +2120,34 @@ def fit_H_and_D(pig, wavelengths, spec, light_fits, CLSF, unc=1, IPH_bounds=(Non
     # Get bin edges in nm.
     edges = get_bin_edges(light_fits)
 
+    # define arguments for objective functions
+    objfn_args = (wavelengths, edges, CLSF, spec, unc, BU_bg, fit_IPH_component)
+    lineshape_model_args = (wavelengths, edges, CLSF, BU_bg, fit_IPH_component)
+    
     # Now call the fitting routine
     if fitter=="scipy":
         # If doing things with the BU background
         if BU_bg is not None:
             pig = pig[0:-2] # skip the modeled background.
 
-        bestfit = sp.optimize.minimize(negloglikelihood, pig, 
-                                       args=(wavelengths, edges, CLSF, spec, \
-                                             unc, BU_bg), 
+        bestfit = sp.optimize.minimize(negloglikelihood, pig,
+                                       args=objfn_args,
                                        method=solver,
-                                       bounds=[(None, None), (None, None), 
-                                               (0, None), 
-                                               (121.55, 121.58), IPH_bounds, 
-                                               (0.0001, 0.02), 
-                                               (None, None), (None, None)]
+                                       bounds=[(None, None), # DN_H
+                                               (None, None),  # DN_D
+                                               (None, None), # DN_IPH
+                                               (121.55, 121.58),  # λ H
+                                               (pig[4]-IPH_wv_spread/2, 
```

Comment:
I think +/- 5 km/s may be safer, this is +/- 2.5 km/s, right?

Reply:
Ah ok, I misunderstood you as meaning 5 for the FWHM

Reply:
Fixed in 2b5aa95a4a3260adfed691da8f079d988d183a70

Reply:
I think you either need to update the comment in constants to refer to 10 km/s or change the constant back and remove the /2 from the bounds array.